### ISSUE TITLE
Handeling 404 status code

### ISSUE THREAD

**Issue by radtrentasei** [CiscoDevNet/ansible-viptela#11]
Sometimes it might happen  that URLs become invalid due to changes in the APIs. 

For example refer to: https://github.com/CiscoDevNet/sdwan-ops/issues/2 

Here what happens is that a previously valid url (/dataservice/template/policy/list/localdomain) become invalid. 

This now leads to a module failure. 

I am suggesting to avoid failing in the module and to offload the responsibility of checking for the right content to other methods. 

As example consider the following ones:
```
get_policy_list_list
get_policy_list_dict
get_policy_definition_list
get_policy_definition_dict
```

**Comment by radtrentasei** [comment]
Pull request generated #12

**Comment by stevenca** [comment]
Addressed with #15
